### ISSUE TITLE
pcre_exec crashes for some regular expressions

### ISSUE THREAD

**Issue by wladston** [samr7/vanitygen#43]
When I try:

```
./vanitygen -r "^1(.)(.)(.).*$"
```

I get the PCRE error: 0. I tried to fix this problem, but I was unable :/

**Comment by BlinkyStitt** [comment]
What pattern is that supposed to match?

**Comment by wladston** [comment]
This pattern is just a simplest crash sample, it's supposed to match anything. One interesting pattern that crashes on the same bug is this one

```
^1(.)(.)(.)(.).?\4\3\2\1[1].*$
```

It would match an address that starts with a palindrome word.

**Comment by BlinkyStitt** [comment]
`^1....*$` works fine

**Comment by wladston** [comment]
Yes, but in order to do complex matchings (such as the palindrome), groups (the parenthesis) are required.
